src/chunked_scatter/chunked_scatter.py

```diff
@@ -0,0 +1,165 @@
+                    out_file = open("{}{}.bed".format(args.prefix,
+                        current_scatter), "w")
+            out_file.write("{}\t{}\t{}\n".format(chunk[0], chunk[1], chunk[2]))
+            current_scatter_size += (chunk[2]-chunk[1])
+        out_file.close()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=
+        "Given a sequence dict or a bed file, scatter over the defined "
+        "contigs/regions. Each contig/region will be split into multiple "
+        "overlapping regions, which will be written to a new bed file."
+        "Each contig will be placed in a new file, unless the length of the "
+        "contigs/regions doesn't exceed a given number.")
+    parser.add_argument("-p", "--prefix", type=str, required=True,
+                        help="The prefix of the ouput files. Output will be "
+                        "named like: <PREFIX><N>.bed, in which N is an "
+                        "incrementing number. This option is mandatory.")
+    parser.add_argument("-i", "--input", type=Path, required=True,
+                        help="The input file, either a bed file or a sequence "
+                        "dict. Which format is used is detected by the "
+                        "extension: '.bed' or '.dict'. This option is "
+                        "mandatory.")
+    parser.add_argument("-c", "--chunk-size", type=int, default=1e6,
+                        metavar="SIZE",
+                        help="The size of the chunks. The first chunk in a "
+                        "region or contig will be exactly length SIZE, "
+                        "subsequent chunks will SIZE + OVERLAP and the final "
+                        "chunk may be anywhere from 0.5 to 1.5 times SIZE "
+                        "plus overlap. If a region (or contig) is smaller "
+                        "than SIZE the original regions will be returned. "
+                        "Defaults to 1e6")
+    parser.add_argument("-m", "--minimum-bp-per-file", type=int, default=45e6,
+                        help="The minimum number of bases represented within "
+                        "a single output bed file. If an input contig or "
+                        "region is smaller than this MINIMUM_BP_PER_FILE, "
+                        "then the next contigs/regions will be placed in the "
+                        "same file untill this minimum is met. Defaults to "
+                        "45e6.")
+    parser.add_argument("-o", "--overlap", type=int, default=150,
+                        help="The number of bases which each chunk should "
+                        "overlap with the preceding one. Defaults to 150.")
+    args = parser.parse_args()
+
+    # Check whether or not the input file is an accepted format.
+    if args.input.suffix == ".bed":
+        bed_input = True
+    elif args.input.suffix == ".dict":
+        bed_input = False
+    else:
+        parser.error("bed or dict input expected, got '{}'".format(
```

Comment:
Is this a parser error? Why not raise a 'ValueError'?
Also it can still be a bed or dict file with a wrong extension. So maybe the message can be something along the lines of `only files with .bed or .dict extension are supported`

Reply:
It's an error in the inputs which the parser parses, but I suppose it's a matter of semantics (does the parser also validate the input or not).